src/analyze/analyze.c

```diff
@@ -1213,7 +1239,99 @@ static int dot(int argc, char *argv[], void *userdata) {
+                { "Style", "BindsTo",           config_parse_string,  0, &arg_dependency_style[UNIT_BINDS_TO]  },
+                { "Style", "PartOf",            config_parse_string,  0, &arg_dependency_style[UNIT_PART_OF]   },
+                { "Style", "OnFailure",         config_parse_string,  0, &arg_dependency_style[UNIT_ON_FAILURE]},
+                { "Style", "Triggers",          config_parse_string,  0, &arg_dependency_style[UNIT_TRIGGERS]  },
+                {}
+        };
+
+        static const char* colors[] = {
+                "black",    /* UNIT_REQUIRES */
+                "blue",     /* UNIT_REQUISITE */
+                "grey66",   /* UNIT_WANTS */
+                "magenta",  /* UNIT_BINDS_TO */
+                "black",    /* UNIT_PART_OF */
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                NULL,
+                "red",      /* UNIT_CONFLICTS */
+                NULL,
+                NULL,
+                "green",    /* UNIT_AFTER */
+                "orange",   /* UNIT_ON_FAILURE */
+                "grey66"    /* UNIT_TRIGGERS */
+        };
+
+        for (UnitDependency dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                if (colors[dependency]) {
+                        arg_dependency_color[dependency] = strdup(colors[dependency]);
+
+                        if (dependency == UNIT_PART_OF)
+                                arg_dependency_style[dependency] = strdup("dashed");
+                        else if (dependency == UNIT_TRIGGERS)
+                                arg_dependency_style[dependency] = strdup("dotted");
+                        else
+                                arg_dependency_style[dependency] = strdup("solid");
+
+                        if (!arg_dependency_color[dependency] || !arg_dependency_style[dependency]) {
+                            return log_oom();
+                        }
+                }
+        }
+
+        /* colors and styles may be overridden on cmd line */
+        STRV_FOREACH(arg_dot_dep_color_style, arg_dot_dependency_colors_styles) {
+                char *ptr = strtok(*arg_dot_dep_color_style, delim);
+                if (ptr != NULL) {
+                        for (UnitDependency dependency = UNIT_REQUIRES; dependency <= UNIT_TRIGGERS; ++dependency) {
+                                if (colors[dependency] &&
+                                    strcmp(unit_dependency_to_string(dependency), ptr) == 0) {  /* unit found */
+                                        ptr = strtok(NULL, delim);
+                                        if (ptr != NULL) {  /* color found */
+                                                free(arg_dependency_color[dependency]);
+                                                arg_dependency_color[dependency] = strdup(ptr);
+
+                                                ptr = strtok(NULL, delim);
```

Comment:
please do not use strtok() ever in our codebase. Use extract_first_word() or so. strtok() is completely broken in potentially threaded code

Reply:
done